### PR TITLE
Edit demographic form for GDPR compliance

### PR DESCRIPTION
**Pull request by kimberscott**
MIT's office of general counsel advised us stop collecting any demographic information from EU participants for the time being. I made some quick changes to the demographic survey so that if someone hasn't selected a non-EU country no personal data is collected. Have tested on staging and it works as intended.

### CODE REVIEW

File: web/templates/web/demographic-data-update.html [hunk 1]
```diff
@@ -12,25 +12,89 @@
...
                 } else {
                     hideField('state');
                     $(`#id_state`)[0].value = '';
                 }
+                // Then show/hide most demographic fields based on EU or not
+				if (gdpr_country_list.includes($('#id_country')[0].value)) {
+					$.each(gdprFieldsToHide, function(i, fieldName) {
+						hideField(fieldName);
+						$(`#id_${fieldName}`)[0].value = '';
+						$(`#id_${fieldName} input[type=checkbox]`).prop("checked", false );
```
Comment (pattisdr) [inline web/templates/web/demographic-data-update.html]: Minor: I would consolidate these three lines into a function that can be called both here and [here](https://github.com/lookit/lookit-api/pull/98/files#diff-b8d298b6a4a06a533d94e07abb55422cR70)

File: web/templates/web/demographic-data-update.html [hunk 2]
```diff
@@ -12,25 +12,89 @@
...
+            
+            if ($('#id_country')[0].value !== 'US') {
+                hideField('state');
+            }
+            
+            if (gdpr_country_list.includes($('#id_country')[0].value)) {
+            	$.each(gdprFieldsToHide, function(i, fieldName) {
+            		hideField(fieldName);
+            		$(`#id_${fieldName}`)[0].value = '';
+            		$(`#id_${fieldName} input[type=checkbox]`).prop("checked", false );
```
Comment (pattisdr) [inline web/templates/web/demographic-data-update.html]: Nice job removing data that might be in this field 👍

File: web/templates/web/demographic-data-update.html [hunk 3]
```diff
@@ -54,10 +119,12 @@
         });
         function hideField(field) {
             $(`#id_${field}`).hide().prop('required',false);
+            $(`#id_${field}`).parent().hide(); //hide the parent form group
             $(`label[for=id_${field}]`).hide();
         }
         function showField(field) {
             $(`#id_${field}`).show();
+            $(`#id_${field}`).parent().show(); // show the parent form group
```
Comment (pattisdr) [inline web/templates/web/demographic-data-update.html]: What is the parent form group?

File: web/templates/web/demographic-data-update.html [hunk 4]
```diff
@@ -12,25 +12,89 @@
 {% block content %}
     <script>
         $(document).ready(function() {
-            if ($('#id_country')[0].value !== 'US') {
-                hideField('state');
-            }
+        
+			const gdpr_country_list = [
```
Comment (pattisdr) [inline web/templates/web/demographic-data-update.html]: Would there be any benefit to pulling this list from a third-party source that updates if this country list changes?  I see you're also adding "", NO, IS, LI, so maybe not.

File: web/templates/web/demographic-data-update.html [hunk 5]
```diff
@@ -12,25 +12,89 @@
...
+            if (gdpr_country_list.includes($('#id_country')[0].value)) {
+            	$.each(gdprFieldsToHide, function(i, fieldName) {
+            		hideField(fieldName);
+            		$(`#id_${fieldName}`)[0].value = '';
+            		$(`#id_${fieldName} input[type=checkbox]`).prop("checked", false );
+            	});
+            }
+            
             $('#id_country').on('change', function() {
+            	// First show/hide state field based on USA or not
```
Comment (pattisdr) [inline web/templates/web/demographic-data-update.html]: Just for my understanding, why do you have to show/hide state field first?

Comment (kimberscott) [reply web/templates/web/demographic-data-update.html]: Sorry, bad comment. What I meant was just do state fields and everything else *separately* for clarity. (It was briefly structured as "are you in the US? if show, show state. If not, are you in the EU? Show/hide personal data fields accordingly." so that fields weren't appropriately showing back up if someone selected US.)

File: web/templates/web/demographic-data-update.html (hunk 4 above)
Comment (kimberscott) [reply web/templates/web/demographic-data-update.html]: Probably, I just didn't find a good source. I suspect as GDPR takes effect this will be a fairly common need and there will be a good option, so I'll keep that in mind.

File: web/templates/web/demographic-data-update.html (hunk 1 above)
Comment (kimberscott) [reply web/templates/web/demographic-data-update.html]: Fixed, thanks!

File: web/templates/web/demographic-data-update.html [hunk 6]
```diff
@@ -119,12 +115,17 @@
         });
         function hideField(field) {
             $(`#id_${field}`).hide().prop('required',false);
-            $(`#id_${field}`).parent().hide(); //hide the parent form group
+            $(`#id_${field}`).parent('.form-group').hide(); // hide the parent form-group div
             $(`label[for=id_${field}]`).hide();
         }
+        function clearField(field) { // hide field and remove any stored data
+            hideField(fieldName);
```
Comment (pattisdr) [inline web/templates/web/demographic-data-update.html]: Drive-by comment, variables need to be consistent.  I would change "field" argument to fieldName.

Comment (kimberscott) [reply web/templates/web/demographic-data-update.html]: Darnit thank you! that's what I get for not testing "trivial" changes locally...
